src/init/features/hosting/github.ts

```diff
@@ -405,16 +405,40 @@ async function promptForRepo(
       message:
         "For which GitHub repository would you like to set up a GitHub workflow? (format: user/repository)",
       validate: async (repo: string) => {
-        // eslint-disable-next-line camelcase
-        const { body } = await githubApiClient.get<{ key: string; key_id: string }>(
-          `/repos/${repo}/actions/secrets/public-key`,
-          {
-            headers: { Authorization: `token ${ghAccessToken}`, "User-Agent": "Firebase CLI" },
-            queryParams: { type: "owner" },
+        try {
+          // eslint-disable-next-line camelcase
+          const { body } = await githubApiClient.get<{ key: string; key_id: string }>(
+            `/repos/${repo}/actions/secrets/public-key`,
+            {
+              headers: { Authorization: `token ${ghAccessToken}`, "User-Agent": "Firebase CLI" },
+              queryParams: { type: "owner" },
+            }
+          );
+          key = body.key;
+          keyId = body.key_id;
+        } catch (e) {
+          if (e.status === 403) {
+            logger.info();
+            logger.info();
+            logWarning("The provided authorization cannot be used with this repository.", "error");
+            logWarning(
+              "If this repository is in an organization, did you remember to grant access?",
+              "error"
+            );
```

Comment:
Minor comment: could you combine these into a single error message?  It's ok if it's long, we have plenty of multi-line error messages.

Reply:
Alright, I updated this to be one error message instead of two, and updated the screenshot in the PR to match.